src/arduino.cc/arduino-builder/main.go

```diff
@@ -249,8 +248,7 @@ func main() {
 		_, err := os.Stat(buildPath)
 		if err != nil {
 			fmt.Fprintln(os.Stderr, err)
-			defer os.Exit(1)
-			return
+			os.Exit(1)
```

Comment:
Another approach may be to make the `main` function just call sub-functions and convert the result into an exit code, something like:

```go
func main() {
    if err := parseCommandLine(); err != nil {
        exitWithError(err);
    }
    if err := runCommands(); err != nil {
        exitWithError(err);
    }
    os.Exit(0)
}

func exitWithError(err error) {
    var exitCode int
    // [...convert error into an exitCode...]
    os.Exit(exitCode)
}
```